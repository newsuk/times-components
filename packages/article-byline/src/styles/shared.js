import styleguide from "@times-components/styleguide";

const { colours, fontFactory } = styleguide();
const shared = {
  link: {
    color: colours.functional.action,
    ...fontFactory({
      font: "supporting",
      fontSize: "cardMeta"
    })
  },
  text: {
    ...fontFactory({
      font: "supporting",
      fontSize: "cardMeta"
    }),
    color: colours.functional.secondary,
    flexDirection: "row"
  }
};

export default shared;
