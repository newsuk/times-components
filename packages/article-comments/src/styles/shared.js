import styleguide from "@times-components/styleguide";

const { colours, fontFactory, spacing } = styleguide();
const styles = {
  button: {
    marginBottom: spacing(0),
    marginTop: spacing(5),
    maxWidth: 215
  },
  container: {
    alignItems: "center",
    borderStyle: "solid",
    borderTopColor: colours.functional.keyline,
    borderTopWidth: 1,
    marginBottom: spacing(5),
    width: "100%"
  },
  headline: {
    color: colours.functional.primary,
    ...fontFactory({
      font: "headline",
      fontSize: "commentsHeadline"
    }),
    maxWidth: 315,
    paddingBottom: spacing(2),
    paddingTop: spacing(6),
    textAlign: "center"
  },
  link: {
    color: colours.functional.action
  },
  supporting: {
    ...fontFactory({
      font: "supporting",
      fontSize: "tertiary"
    }),
    color: colours.functional.secondary,
    maxWidth: 280,
    textAlign: "center"
  }
};

export default styles;
