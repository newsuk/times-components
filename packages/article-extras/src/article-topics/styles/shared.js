import { colours, tabletWidth } from "@times-components/styleguide";

const sharedStyles = {
  topicsContainer: {
    borderTopColor: colours.functional.keyline,
    borderTopWidth: 1
  },
  topicsContainerTablet: {
    alignSelf: "center",
    width: tabletWidth
  },
  topicsMetaContainer: {
    justifyContent: "flex-start"
  }
};

export default sharedStyles;
