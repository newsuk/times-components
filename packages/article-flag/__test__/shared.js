/* eslint-env jest */

import "react-native";
import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";

import ArticleFlag, {
  NewArticleFlag,
  UpdatedArticleFlag,
  ExclusiveArticleFlag,
  SponsoredArticleFlag
} from "../article-flag";

module.exports = () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ArticleFlag title="articleFlag" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders New flag correctly", () => {
    const tree = renderer.create(<NewArticleFlag />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders Updated flag correctly", () => {
    const tree = renderer.create(<UpdatedArticleFlag />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders Exclusive flag correctly", () => {
    const tree = renderer.create(<ExclusiveArticleFlag />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders Sponsored flag correctly", () => {
    const tree = renderer.create(<SponsoredArticleFlag />).toJSON();
    expect(tree).toMatchSnapshot();
  });
};
