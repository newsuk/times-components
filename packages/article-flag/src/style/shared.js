import styleguide from "@times-components/styleguide";

const { fontFactory } = styleguide();
const styles = {
  diamond: {
    marginRight: 5
  },
  title: {
    ...fontFactory({
      font: "bodyRegularSmallCaps",
      fontSize: "cardMetaMobile"
    }),
    fontWeight: "400",
    letterSpacing: 1.4
  },
  view: {
    alignItems: "center",
    flexDirection: "row"
  }
};

export default styles;
