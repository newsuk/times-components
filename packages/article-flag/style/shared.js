import { fonts } from "@times-components/styleguide";

const styles = {
  view: {
    flexDirection: "row",
    alignItems: "center"
  },
  diamond: {
    marginRight: 5,
    marginBottom: 1
  },
  title: {
    fontFamily: fonts.bodyRegularSmallCaps,
    fontSize: 12,
    fontWeight: "400",
    letterSpacing: 1.4
  }
};

export default styles;
