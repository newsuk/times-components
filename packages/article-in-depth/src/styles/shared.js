import styleguide from "@times-components/styleguide";

const { colours, fontFactory, spacing } = styleguide();
const sharedStyles = {
  articleHeadline: {
    ...fontFactory({
      font: "headline",
      fontSize: "headline"
    }),
    color: colours.functional.brandColour,
    marginBottom: spacing(2),
    textAlign: "center"
  },
  container: {
    alignItems: "center",
    paddingBottom: spacing(9),
    paddingHorizontal: spacing(4),
    paddingTop: spacing(12)
  },
  datePublication: {
    ...fontFactory({
      font: "supporting",
      fontSize: "cardMeta"
    }),
    color: colours.functional.secondary,
    marginTop: spacing(1)
  },
  flag: {
    flexDirection: "row",
    marginBottom: spacing(3),
    marginTop: spacing(1)
  },
  flagContainer: {
    marginRight: spacing(3)
  },
  label: {
    marginBottom: spacing(2)
  },
  leadAssetContainer: {
    marginBottom: spacing(4)
  },
  meta: {
    flexDirection: "row",
    flexWrap: "wrap"
  },
  metaContainer: {
    alignItems: "center",
    borderBottomWidth: 1,
    borderColor: colours.functional.keyline,
    borderTopWidth: 1,
    marginBottom: spacing(5),
    marginHorizontal: spacing(2),
    paddingBottom: 6,
    paddingTop: spacing(2)
  },
  standFirst: {
    ...fontFactory({
      font: "headlineRegular",
      fontSize: "smallestHeadline"
    }),
    color: colours.functional.primary,
    marginBottom: spacing(3),
    paddingHorizontal: spacing(2),
    textAlign: "center"
  }
};

export default sharedStyles;
