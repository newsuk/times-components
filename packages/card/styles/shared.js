import { StyleSheet } from "react-native";

export default StyleSheet.create({
  imageContainer: {
    marginBottom: 10
  },
  headerContainer: {
    height: 24,
    marginBottom: 10,
    maxWidth: 300
  },
  textContainer: {
    height: 10,
    marginBottom: 10
  },
  lastBar: {
    marginBottom: 0,
    maxWidth: 240
  }
});
