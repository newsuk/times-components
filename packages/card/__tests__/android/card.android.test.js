import shared from "../shared";

describe("Card tests on android", () => {
  shared();
});
