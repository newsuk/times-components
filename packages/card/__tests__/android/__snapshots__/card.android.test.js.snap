// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card tests on android renders loading state 1`] = `
<View>
  <View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <View
        aspectRatio={1.5}
        style={Object {}}
      >
        <View
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <Image
            onLoad={[Function]}
            source={
              Object {
                "uri": "&preview=true",
              }
            }
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": "100%",
                  "width": "100%",
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          >
            <Image
              onLoad={[Function]}
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": "100%",
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            />
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "flex": 1,
                  },
                  null,
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <BVLinearGradient
                  colors={
                    Array [
                      "#F9F9F9",
                      "#EDEDED",
                    ]
                  }
                  end={
                    Object {
                      "x": -0.09302964577578221,
                      "y": 0.11488204504197774,
                    }
                  }
                  locations={
                    Array [
                      0,
                      1,
                    ]
                  }
                  start={
                    Object {
                      "x": 1,
                      "y": 0,
                    }
                  }
                  style={
                    Array [
                      Object {
                        "flex": 1,
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View>
      <View
        style={
          Array [
            Object {
              "height": 24,
              "marginBottom": 10,
              "maxWidth": 300,
            },
          ]
        }
      >
        <BVLinearGradient
          colors={
            Array [
              "#F9F9F9",
              "#EDEDED",
            ]
          }
          end={
            Object {
              "x": -0.09302964577578221,
              "y": 0.11488204504197774,
            }
          }
          locations={
            Array [
              0,
              1,
            ]
          }
          start={
            Object {
              "x": 1,
              "y": 0,
            }
          }
          style={
            Array [
              Object {
                "flex": 1,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "height": 10,
              "marginBottom": 10,
            },
          ]
        }
      >
        <BVLinearGradient
          colors={
            Array [
              "#F9F9F9",
              "#EDEDED",
            ]
          }
          end={
            Object {
              "x": -0.049525179317961854,
              "y": 0.05500328394968961,
            }
          }
          locations={
            Array [
              0,
              1,
            ]
          }
          start={
            Object {
              "x": 1,
              "y": 0,
            }
          }
          style={
            Array [
              Object {
                "flex": 1,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "height": 10,
              "marginBottom": 10,
            },
          ]
        }
      >
        <BVLinearGradient
          colors={
            Array [
              "#F9F9F9",
              "#EDEDED",
            ]
          }
          end={
            Object {
              "x": -0.049525179317961854,
              "y": 0.05500328394968961,
            }
          }
          locations={
            Array [
              0,
              1,
            ]
          }
          start={
            Object {
              "x": 1,
              "y": 0,
            }
          }
          style={
            Array [
              Object {
                "flex": 1,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "height": 10,
              "marginBottom": 10,
            },
            Object {
              "marginBottom": 0,
              "maxWidth": 240,
            },
          ]
        }
      >
        <BVLinearGradient
          colors={
            Array [
              "#F9F9F9",
              "#EDEDED",
            ]
          }
          end={
            Object {
              "x": -0.049525179317961854,
              "y": 0.05500328394968961,
            }
          }
          locations={
            Array [
              0,
              1,
            ]
          }
          start={
            Object {
              "x": 1,
              "y": 0,
            }
          }
          style={
            Array [
              Object {
                "flex": 1,
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`Card tests on android renders vertical by default 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <View
        aspectRatio={1.5}
        style={Object {}}
      >
        <View
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <Image
            onLoad={[Function]}
            source={
              Object {
                "uri": "https://www.thetimes.co.uk/imageserver/image/methode%2Fsundaytimes%2Fprod%2Fweb%2Fbin%2F9242e576-4dfc-11e7-a20e-a11097d3353d.jpg?crop=1463%2C975%2C293%2C12&resize=360&preview=true",
              }
            }
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": "100%",
                  "width": "100%",
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Object {
                "height": "100%",
                "width": "100%",
              }
            }
          >
            <Image
              onLoad={[Function]}
              source={
                Object {
                  "uri": "https://www.thetimes.co.uk/imageserver/image/methode%2Fsundaytimes%2Fprod%2Fweb%2Fbin%2F9242e576-4dfc-11e7-a20e-a11097d3353d.jpg?crop=1463%2C975%2C293%2C12&resize=360",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": "100%",
                    "width": "100%",
                  },
                  undefined,
                ]
              }
            />
            <View
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "flex": 1,
                  },
                  null,
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <BVLinearGradient
                  colors={
                    Array [
                      "#F9F9F9",
                      "#EDEDED",
                    ]
                  }
                  end={
                    Object {
                      "x": -0.09302964577578221,
                      "y": 0.11488204504197774,
                    }
                  }
                  locations={
                    Array [
                      0,
                      1,
                    ]
                  }
                  start={
                    Object {
                      "x": 1,
                      "y": 0,
                    }
                  }
                  style={
                    Array [
                      Object {
                        "flex": 1,
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Some text
      </Text>
    </View>
  </View>
</View>
`;

exports[`Card tests on android renders without image 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        No image
      </Text>
    </View>
  </View>
</View>
`;

exports[`Card tests on android renders without image url 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "opacity": 0,
    }
  }
>
  <View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        No image URL
      </Text>
    </View>
  </View>
</View>
`;
